### ISSUE TITLE
External Video Monitoring (Security System)

### ISSUE THREAD

**Issue by nelsonic** [dwyl/home#130]
# Why? (Context)

Currently our front [external] gate uses a simple latch (lock) which can be opened by _anyone_. 
e.g. the postman (person) opens the gate to enter the property and place mail in the box. 
This has been OK while the house has been in development, but it's not "secure".
It means _anyone_ can enter the front gate and access the garden/outside of the house. 
Once we have the door entry system https://github.com/dwyl/smart-home-security-system/issues/1 working,
we will implement secure access for the front gate by upgrading the existing latch to be electronic. 
However having a more secure access system on the external gate is merely a deterrent. 
It _discourages_ unwanted access to the property, but it does not _prevent_ it. 
The only way to _prevent_ access is higher walls covered in razor blade wire, but that would make our house a _prison_! 🤦 

![image](https://user-images.githubusercontent.com/194400/86737257-0dbff780-c02c-11ea-9899-87e9fd5519bb.png)

_Obviously_ none of us wants to have high walls or razor wire.  We don't want to live in a [dystopian](https://en.wikipedia.org/wiki/Dystopia) ["walled garden"](https://en.wikipedia.org/wiki/Closed_platform). 
So instead we want to _monitor_ if people attempt to access the property using a camera system.

# What? 

A security camera and recording system for the _outside_ of the property so we can monitor attempted access.
A basic alerting system that should send a message to the house master and display the captured images on "info" screens in the house. If nobody attends to the intrusion/alert within 30 seconds, sound a loud alarm both inside and outside the house.

We can approach this in two ways:
1. Build our own system from scratch using off-the-shelf components (Raspberry Pi, Pi Cam (or USB WebCam) + 3D Printed enclosure) 

[![image](https://user-images.githubusercontent.com/194400/86765067-921c7580-c040-11ea-9de9-3d4e4e5698d7.png)](https://youtu.be/og-zI0CozLs)
https://youtu.be/og-zI0CozLs

2. Buy a high quality security monitoring solution from 

The company/products I would considering is UniFi Protect from Ubiquiti: https://unifi-network.ui.com/video-security
![image](https://user-images.githubusercontent.com/194400/86733380-0f3bf080-c029-11ea-9fa3-d49e34ec31b0.png)

![image](https://user-images.githubusercontent.com/194400/86740304-7314e800-c02e-11ea-96ca-51d83f8d8272.png)

We have already invested in UniFi Switches and Access Points (APs) so I think this is a fairly logical choice. 💭 
I would _never_ consider a "Cloud" based camera system because (a) I don't want my movements tracked by a random Faceless corporation (and neither should you!!) and (b) I don't want to pay anyone a monthly ***rent*** for my security. I want a one-off payment and then to ***own*** the system even if that one-off up-front payment seems higher on the surface. I consider the useful life of the system to be 5 years. There's no reason it should not continue working for longer than that. 

Further research required. 🔍

**Comment by nelsonic** [comment]
The **UniFi Protect G3 FLEX Camera** is the cheapest security camera they make. 
It has **1080p** resolution, night vision (IR) and motion detection.
https://store.ui.com/collections/unifi-protect/products/unifi-video-g3-flex-camera
![image](https://user-images.githubusercontent.com/194400/103179786-69569980-4887-11eb-9d4f-e7c9c0e665c5.png)

The Crosstalk video has a good side-by-side comparison with the G3 Bullet Camera, the have the same quality.
[![image](https://user-images.githubusercontent.com/194400/103180062-2e099a00-488a-11eb-9382-595c3e227379.png)](https://youtu.be/f6zITSH7XNY0)

[![image](https://user-images.githubusercontent.com/194400/103181173-294ae300-4896-11eb-886b-a3b6b003287d.png)](https://youtu.be/a_RRxfh8WPs)

I think this is a good starting point along with the  https://github.com/dwyl/home/issues/169

**Comment by nelsonic** [comment]
We have 5 external entrance points: 
1. Front door 
2. Kitchen
3. Basement
4. Leisure Room 
5. Garage

I think we can justify buying the **3-pack** of G3 Flex Cameras. 💭 
https://www.mbit.pt/ubiquiti-unifi-video-camera-3-pack-uvc-g3-flex-3
![image](https://user-images.githubusercontent.com/194400/103181280-724f6700-4897-11eb-9858-c00635d91b74.png)

it's only a €10 discount, but ultimately, I think we will have **at _least_ 6 cameras** to cover the exterior of the house.

**Comment by nelsonic** [comment]
After reviewing the location of the network ports at the front, kitchen and basement doors I've noticed that the _previous_ electricians did not follow our instructions for their placement. This means we will need to make significant holes in the walls in order to run the Cat5 cables to the desired locations. 🤦  This will take several days work to make the "rasgos", place the cables and then patch up the walls again. 

There is a work-around: the UniFi Protect G3 Micro Camera can connect via Wifi and power via 220V (using an adapter)
So we would be able to have cameras wherever there are electrical points i.e. Lights.
The downside is that the G3 Micro Cameras cost **$199 _each_** ... 💸 

https://store.ui.com/collections/unifi-protect/products/unifi-video-g3-micro
![image](https://user-images.githubusercontent.com/194400/103182253-04f50380-48a2-11eb-9de5-ff4d210c5e7f.png)

Cheapest in PT is €200: https://www.senetic.pt/product/UVC-G3-MICRO
![image](https://user-images.githubusercontent.com/194400/103182087-79c73e00-48a0-11eb-85ce-2a474d8f55b9.png)

Bit annoyed at this ... 😕

**Comment by nelsonic** [comment]
Ubiquiti _just_ released the **G3 Instant** which is a $29 Wifi/USB-C Camera (_less than **half** the price of G3 Flex!_):
https://eu.store.ui.com/collections/unifi-protect-cameras/products/unifi-protect-g3-instant-camera
![UVC-G3-INS](https://user-images.githubusercontent.com/194400/104451511-900c1400-5599-11eb-955e-59b9210f074d.png)

Obviously it's immediately **sold out**. And when it was available it was limited to **1 per** retail **customer** ... 🙄 

Crosstalk has a good review/comparison:
[![image](https://user-images.githubusercontent.com/194400/104451108-02302900-5599-11eb-8c76-e7110c17d733.png)](https://youtu.be/JmLqZ36aKJA)

The best thing about this new camera is the wider angle (field of view) **133 Degrees**:

![image](https://user-images.githubusercontent.com/194400/104452241-9e0e6480-559a-11eb-93d8-9eeb203382b2.png)

It's very much an ***indoor*** camera (no IP rating or weather protection).

I think this camera is _perfect_ in the instances where the original electricians did not follow our instructions/plan for the placement of Cat6 (network) cable and therefore we would still need to extend the cable if we use an IP camera.
This is **_excellent_ news** and will lower our costs for _most_ of the cameras we need. 
There are still a few instances where we may _prefer_ not to have a bulky power adapter on show and already have Cat6 in which case the g3 Flex is a better option. 

I'm _very_ glad I didn't pony up for the G3 Micro ... waiting for CES has paid off! 🙌

**Comment by iteles** [comment]
Excellent news! Let's hope they become available to us soon 🤞

**Comment by nelsonic** [comment]
@iteles yeah, I have signed up for notifications and will buy as soon as they are available, but not holding my breath. ⏳

**Comment by nelsonic** [comment]
After watching a few more videos, I decided we _definitely_ didn't need the Dream Machine 
because we already have a 16-port POE Switch in our network and are only using 4 of the ports. 
When we bought the switch we already planned on having the cameras and access points attached.

We need _something_ to record the video, so I got the "_basic_" Cloud Key Gen2 Plus **`SKU:  UCK-G2-PLUS`**:

https://eu.store.ui.com/collections/unifi-accessories-cloud-key/products/unifi-cloudkey-gen2-plus
![image](https://user-images.githubusercontent.com/194400/166824091-fcb1af9b-ec07-4b9a-a5b4-98169d1aa1cb.png)

![image](https://user-images.githubusercontent.com/194400/166823891-f47d788a-e6d6-4fc3-8a19-0fe8b3313a2f.png)

Review: Cloud Key Gen2 Plus https://youtu.be/H_a1BCf1jH0
[![image](https://user-images.githubusercontent.com/194400/166824993-0801d57d-d588-44a6-800a-b77f028c6c19.png)](https://youtu.be/H_a1BCf1jH0)

Order placed: `Order #NL-EU330019`

![image](https://user-images.githubusercontent.com/194400/166823062-63eed10c-3f16-417d-8a93-a2fe7ac2d086.png)

The advantage of buying it directly from Ubiquiti is that they don't charge us the VAT. 

Included the Doorbell https://github.com/dwyl/home/issues/201 and _Basic_ G3 Camera (_which works over WIFI_) in the order. See above for product detail. Additionally, added a "Access Point WiFi 6 Long-range" **`SKU: U6-LR`**

https://eu.store.ui.com/collections/unifi-network-wireless/products/unifi-6-long-range-access-point-1
![image](https://user-images.githubusercontent.com/194400/166824459-3dba9613-5fcc-4291-b5d1-1559e3214156.png)

This should boost the network signal/range/bandwidth.
I will be installing on the ceiling of the large room and testing if it gives people in the garden decent wifi signal.
The APs we have currently are **`UAP-AC-Pro`**: https://eu.store.ui.com/products/unifi-ac-pro

![image](https://user-images.githubusercontent.com/194400/166825376-9bf62dfc-a052-4b6f-9c2f-c7ec3885d4bd.png)

The Wifi is _usually_ reliable in most of the house. But signal strength is poor in the garden.
First world problems, I know! But if we can boost it and guarantee that both the Doorbell and Camera have uninterrupted signal then it's a win.

**Comment by iteles** [comment]
You have done infinite amounts more research on this so I trust your judgement 👍 
Did you add in the company NIF in the details?

Good call also on the long range AP for the garden/front door wifi range.

**Comment by nelsonic** [comment]
Yeah, used the PT Co NIF & paid from the account. Can forward you the invoice once the equipment arrives. ⏳

**Comment by nelsonic** [comment]
The **`Access Reader Pro`** has been **Sold Out** all year:
https://eu.store.ui.com/collections/unifi-door-access/products/unifi-access-reader-pro
![image](https://user-images.githubusercontent.com/194400/206929237-b548bcaa-b33a-437c-888d-c7965c2d3774.png)

This is the reader with built-in camera and intercom that I want on the front door. 
The touch-screen allows for a one-time key-code fallback if people don't have their card.
I have asked Ubiquiti to notify me when they come back into stock but that was in ***`January 2022`** and I haven't had an in-stock alert all year ... ⌛ 

I have a watcher for this on ebay and one is available at a slight premium but we could get it installed like ASAP!!
https://www.ebay.co.uk/itm/314253263100
<img width="622" alt="image" src="https://user-images.githubusercontent.com/194400/206929167-2b1a8451-5f92-4ce1-b6e8-57efa0572e8e.png">

@iteles given that this is quite a lot of cash for a glorified card access reader
(that is secure and has a built-in camera etc ...) 
Should I pay a premium and get it installed in January so we can get the door entry system working ASAP?

**Comment by nelsonic** [comment]
For reference, there are other sellers charging a _substantially_ higher premium for these due to out-of-stock ...
https://www.ebay.co.uk/itm/185690320369
<img width="774" alt="image" src="https://user-images.githubusercontent.com/194400/206929788-a82c3f91-aba0-4899-8ff6-65138d58a777.png">

**Comment by iteles** [comment]
Yes, I think £30-40 premium is totally worth it to get it done 👌🏻

**Comment by nelsonic** [comment]
I just have no idea when these will be back in-stock. 🤷‍♂️ 
But equally ... we don't _need_ this in order to move forward, we can "just" use the more basic one:
https://eu.store.ui.com/collections/unifi-door-access/products/unifi-access-reader-lite
![image](https://user-images.githubusercontent.com/194400/206937200-758321a3-be08-440f-870f-ec0cc8587737.png)

This is the ultra-basic access reader we will be using at the _back_ of the house for the two less-frequently used doors.
I mean if I had _unlimited_ funds I'd use the "Pro" readers _everywhere_ 

This is suggested network diagram/config:

![image](https://user-images.githubusercontent.com/194400/206937396-7b91f72e-0338-4836-b46a-209ed254c04f.png)

Having the UA-Pro adds polish to the system but it's definitely not _neeeeeded_. 💭

**Comment by nelsonic** [comment]
Still sold out: 
<img width="1325" alt="ua-pro-still-sold-out" src="https://user-images.githubusercontent.com/194400/233779624-b3387092-bc8d-417d-8612-1e5f379c57ff.png">

Checked ebay and there are a couple of reasonably priced units: 
<img width="842" alt="image" src="https://user-images.githubusercontent.com/194400/233779670-ecf61c91-6136-4545-89dc-5abd56082250.png">

<img width="1496" alt="image" src="https://user-images.githubusercontent.com/194400/233779751-734031cf-386b-462e-8b76-e63b6bb326b2.png">

But when I try to checkout the cost jumps:
<img width="1250" alt="ua-pro-price-gouge-checkout-ebay" src="https://user-images.githubusercontent.com/194400/233780090-618a8c43-8aac-41eb-8151-166d41e409c0.png">

Think it might be adding `VAT` ... but doesn't make any sense given it's based in Italy ... 
Meanwhile, the Unifi Access Starter Kit https://github.com/dwyl/learn-computer-networking/issues/15 is back in-stock.
Which _includes_ a `UA-Pro` and the kit (bundle) is _cheaper_ than the sum of the parts.

**Comment by iteles** [comment]
@nelsonic Please send me the invoice for this.

**Comment by nelsonic** [comment]
@iteles you have the login. But yes, I will dig it out when I'm next at my desk. 👌

**Comment by asntc** [comment]
Side-by-side comparison of the G3 instant and G4 instant:

<img width="1919" alt="Screenshot 2024-11-23 at 05 42 43" src="https://github.com/user-attachments/assets/148b4ef8-aa24-499b-9902-db7ad3a9bcd8">

I don't think it's worth paying the €20 extra for the higher resolution.
But we do want another camera for _outdoors_ (back porch). 
Let's wait for BlackFriday to see if there is a discount.
But even without a discount, we should order _three_. 3️⃣ 

https://eu.store.ui.com/eu/en/category/all-cameras-nvrs/products/uvc-g3-ins
https://eu.store.ui.com/eu/en/category/cameras-compact/products/uvc-g4-ins

**Comment by asntc** [comment]
Other products we are considering:

Access Ultra UA-Ultra
https://eu.store.ui.com/eu/en/category/all-door-access/products/ua-ultra

<img width="716" alt="image" src="https://github.com/user-attachments/assets/982b3004-4423-485c-b2e6-4676a731afae">

Really happy Ubiquiti has created this product! 🎉 
It's _exactly_ what we need for the bedrooms @home 🚪 🔐 

Electric Lock UA-Lock-Electric
https://eu.store.ui.com/eu/en/products/ua-lock-electric

<img width="727" alt="image" src="https://github.com/user-attachments/assets/d527ed23-2b2c-48e6-ae56-588dbe5bae8f">

Pocket Keyfob, 10-Pack UA-Pocket
https://eu.store.ui.com/eu/en/products/ua-pocket

<img width="734" alt="image" src="https://github.com/user-attachments/assets/cf8f5859-7778-47bf-bb1c-a9516ce4520f">

Access Button UA-Button
https://eu.store.ui.com/eu/en/category/door-access-readers/products/ua-button?variant=ua-button

<img width="745" alt="image" src="https://github.com/user-attachments/assets/53d58c94-68ed-4829-a26d-2653da4a9688">

_Verrrry_ easy to burn a _lot_ of cash on this company!!

<img width="1003" alt="image" src="https://github.com/user-attachments/assets/9956685f-a978-4547-9c55-a7e563f34089">

But these products aren't for "consumers", they are for biz!
And having _reliable_ & secure products is worth every penny.

**Comment by nelsonic** [comment]
Black Friday Promo for the next 5 days:

<img width="1096" alt="image" src="https://github.com/user-attachments/assets/f1d7d12d-7f51-44db-9b92-21a8d0d63f48">

**Comment by nelsonic** [comment]
<img width="1209" alt="image" src="https://github.com/user-attachments/assets/e4fee69d-1625-4655-b9f8-5c92f7538184">

Got: 
4 x `G3 Instant` cameras for specific spots. 📷 

Half price. 🎉 
<img width="1121" alt="image" src="https://github.com/user-attachments/assets/b2902600-bf6a-452d-935c-2899a49bcd1d">


Sadly, none of the door access stuff was on `Black Friday` promo ... 😢 
5 x `Access Ultra` for the doors (though) might end up needing to buy a couple more ... 💸 🚪 
2 x `Access Button` for Kitchen and Basement doors

And splurged on an `AmpliFi Alien Router` which we will use to get `WiFi` out to the `Secret Garden`
https://eu.store.ui.com/eu/en/category/all-amplifi-alien/collections/amplifi-alien/products/amplifi-alien?variant=afi-aln-r-eu

<img width="1128" alt="image" src="https://github.com/user-attachments/assets/e6846b94-4533-4307-b5a5-1074be2b78be">

The new router will be mounted in the `Dining Room` and serve the Editing Macs with _wired_ Gigabit Ethernet (instead of `WiFi` which maxes at `300mbps`) useful when uploading or just having better connection during video calls. 👌
